index.bs

```diff
@@ -101,21 +118,190 @@ spec: webidl
 </section>
 <section>
   <h2 id="permission-descriptor">
-    Permission descriptor
+    Descriptions of permission requests
   </h2>
   <pre class='idl' title=''>
     dictionary PermissionDescriptor {
       required PermissionName name;
     };
   </pre>
   <p>
-    A permission is described by a name and other properties that depend on the
-    name. The simplest permissions require only a name, but some others have
-    more detailed structure that requires more information to describe it. In
-    that case, they should define a customized <a>permission descriptor
-    type</a> dictionary that inherits from {{PermissionDescriptor}}.
+    Each powerful feature has one or more aspects that websites may request
+    permission to access. To describe these aspects, each feature defines a
+    subtype of {{PermissionDescriptor}} to be its <a>permission descriptor
+    type</a>.
   </p>
+
+  <div class="example" id="example-descriptors">
+    <p>
+      The {{"midi"}} feature has two aspects: access to normal messages, and
+      access to system exclusive messages. Thus, its permission descriptor type
+      is:
+    </p>
+    <pre highlight="idl">
+      dictionary MidiPermissionDescriptor : PermissionDescriptor {
+        boolean sysex = false;
+      };
+    </pre>
+    <p>
+      The {{"camera"}} feature lets sites access whatever cameras are connected
+      to the user's device. Thus, its descriptor type is:
+    </p>
+    <pre highlight="idl">
+      dictionary DevicePermissionDescriptor : PermissionDescriptor {
+        DOMString deviceId;
+      };
+    </pre>
+    <p>
+      General access to cameras is represented by `{name: 'camera'}`, while
+      access to a particular camera is represented by `{name: 'camera',
+      deviceId: cameraId}`.
+    </p>
+  </div>
+</section>
+<section>
+  <h2 id="permission-operations">Permission states</h2>
+  <p>
+    The user agent is responsible for tracking what powerful features each
+    <a>realm</a> has the user's permission to use. Other specifications can use
+    the operations defined in this section to retrieve the UA's notion of what
+    permissions are granted or denied, and to ask the user to grant or deny more
+    permissions.
+  </p>
+
+  <p>
+    Other specifications can also add more constraints on the UA's behavior in
+    these algorithms.
+  </p>
+
+  <p>
+    Within this section, |descriptor| is an instance of the <a>permission
+    descriptor type</a> of the powerful feature named by
+    <code>|descriptor|.{{PermissionDescriptor/name}}</code>.
+  </p>
+
+  <section>
+    <h3 id="reading-current-states">Reading the current permission state</h3>
+    <p>
+      |descriptor|'s <dfn export>permission state</dfn> is one
+      of {{"granted"}}, {{"prompt"}}, or {{"denied"}}, indicating respectively
```

Comment:
maybe link to the enum itself? Otherwise, we will need to duplicate these values if we need to update the enum. If these are the formal definitions for each of these states, then we should break them up into a `dl`.